canonical-web-and-design/maas-ui
# feat(ui): move 'Add machine' + 'Add chassis' forms to machine list header

## Done

- Moved "Add machine" and "Add chassis" forms to machine list header
- Removed old routes
- Added `MachineHeaderForms` component, which houses `AddChassis`, `AddMachine` and `ActionFormWrapper`.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine list and click "Add hardware" > "Machine"
- Check that the title changes to "Add machine" and the form shows up
- Check that you can submit the form via "Save machine" (closes the form on success) and "Save and add another" (does not close form on success)
- Try to submit the form with "Save machine" with an error (e.g. duplicate MAC address) and check that the form does not close
- Check the same things with the "Add chassis" form 

## Fixes

Fixes canonical-web-and-design/app-squad#241

## Screenshot
![Screenshot 2021-09-07 at 14-34-53 Machines bolla MAAS](https://user-images.githubusercontent.com/25733845/132284898-3e70083d-30eb-474f-a0cb-1c2fb80f1c80.png)
